virtcontainers/agent.go

```diff
@@ -152,6 +152,12 @@ type agent interface {
 	// get agent url
 	getAgentURL() (string, error)
 
+	// get the agent's client
+	getClient() interface{}
+
+	// set to use an existing agent's client
+	setClient(client interface{}) error
```

Comment:
What about `reuseAgent(a agent) error` API and internally assert agent is of the right type and reuse agent.client internally. Then we don't expose `interface{}` and agent's internal knowledge around.

Reply:
@bergwolf Updated!